server/src/MessageHandler.ts

```diff
@@ -41,41 +128,34 @@ export async function handleDefinitionRequest(
     return new Promise((resolve) => resolve())
   }
 
-  const documentText = document.getText()
-
   const word = getWordAtPosition(document, position)
+
   if (word === '') {
     return new Promise((resolve) => resolve())
   }
 
-  // parse schem file to datamodel meta format (DMMF)
-  const dmmf = await getDMMF({ datamodel: documentText })
-
-  const modelName = dmmf.datamodel.models
-    .map((model) => model.name)
-    ?.find((name) => name === word)
-
-  // selected word is not a model type
-
-  if (!modelName) {
-    return new Promise((resolve) => resolve())
-  }
+  const documentText = document.getText()
+  const modelName = getWordAtPosition(document, position)
 
   const modelDefinition = 'model '
   // get start position of model type
   const index = documentText.indexOf(modelDefinition + modelName)
   const buf = documentText.slice(0, index)
   const EOL = '\n'
   const lines = buf.split(EOL).length - 1
-  const lastLineIndex = buf.lastIndexOf(EOL)
+  const modelBlock = getBlockAtPosition(lines, document)
+
+  if (!modelBlock) {
+    return new Promise((resolve) => resolve())
```

Comment:
Not sure why TS is allowing this, but if you're trying to return nothing from a function that expects `Promise<Location>`, you should probably change the signature to `Promise<Location|void>`

and then this can be `return new Promise((resolve) => resolve())` => `return`

Reply:
I can change the signature to `Promise<Location|undefined>` but not `Promise<Location|void`, because the function that calls this expects: `HandlerResult<Location | Location[] | LocationLink[] | null | undefined, void>`.

Reply:
Ah sounds good then!